src/js/select2/data/select.js

```diff
@@ -203,6 +203,14 @@ define([
     data = $.data($option[0], 'data');
 
     if (data != null) {
+      // always update the value of `disabled` property of the option element
+      data.disabled = $option.prop('disabled');
```

Comment:
Ideally the solution to this should be very similar, if not the same as, the solution to us handling the cached `selected` property on elements.

https://github.com/select2/select2/commit/54154c3df6b7e2f59dc76b663f230f9b773397f2

Reply:
We now have a PR that is similar to the mentioned commit: #4818

Reply:
@denilsonsa I don't think that #4818 handles `optgroup` elements.  You'll notice that @letrunghieu's solution does a bit more.  However, neither of these PRs have any tests yet 😦